### ISSUE TITLE
end2end_tests: ProxyEnd2end/ProxyEnd2endTest.HugeResponse is flaky (The test stalls)

### ISSUE THREAD

**Issue by sreecha** [grpc/grpc#5018]
**Summary**
The ProxyEnd2endTest.HugeResponse test is flaky. The test succeeds sometimes and stalls sometimes. This feels like an underlying issue in core that is worth exploring.

_Note:_ The test case tests for server sending responses in four scenarios (with and without proxy server and with and without tls). The flakiness happens in ALL cases - i.e not just when using a proxy server.

**Steps to Repro:**

1. Build the sync end 2 end tests: 
 `<grpc_root_dir>$ make CONFIG=dbg end2end_test -j`

2. Run the ```ProxyEnd2endTest.HugeResponse``` test without using proxy or tls (i.e run the first test scenario). (This runs the test 10 times to reproduce the flake)
  `<grpc_root_dir>$  for i in {1..10} ; do bins/opt/end2end_test --gtest_filter="ProxyEnd2end/ProxyEnd2endTest.HugeResponse/0"; done`
 
3. You will notice that the test will stall. 

_Note:_ Interestingly, when I turn on `flowctl` trace, this does not reproduce (it might reproduce - but I could not reproduce it so far with flowctl trace)..i.e the following command does not reproduce the issue! (any other trace flags like api, transport, etc do reproduce the issue). 

```
<grpc_root_dir>$ for i in {1..10} ; do GRPC_TRACE=flowctl bins/opt/end2end_test --gtest_filter="ProxyEnd2end/ProxyEnd2endTest.HugeResponse/0"; done
```

**Comment by atoader** [comment]
Is it possible that this issue is related to https://github.com/grpc/grpc/issues/5037 ?

**Comment by yang-g** [comment]
This should be fixed by #5114.
